src/scifem/biomedical.py

```diff
@@ -0,0 +1,118 @@
+import dolfinx
+import basix.ufl
+from pathlib import Path
+import numpy as np
+import numpy.typing as npt
+
+
+def apply_mri_transform(
+    path: Path, coordinates: npt.NDArray[np.floating], use_tkr: bool = False
+) -> npt.NDArray[np.inexact]:
+    """Given a path to a set of MRI voxel data, return the data evaluated at the given coordinates.
+
+    Args:
+        path: Path to the MRI data, should be a file format supported by nibabel.
+        coordinates: Coordinates to evaluate the MRI data at.
+
+    Returns:
+        The data evaluated at these points.
+    """
+    try:
+        import nibabel
+        import nibabel.affines as naff
+    except ImportError:
+        raise ImportError("This function requires the nibabel package to be installed")
+    image = nibabel.load(path)
+    data = image.get_fdata()
+    # This depends on how one uses FreeSurfer
+    if use_tkr:
+        vox2ras = image.header.get_vox2ras_tkr()
+    else:
+        # VOX to ras explanation: https://surfer.nmr.mgh.harvard.edu/ftp/articles/vox2ras.pdf
+        vox2ras = image.header.get_vox2ras()
+    ras2vox = np.linalg.inv(vox2ras)
+    ijk_vectorized = naff.apply_affine(ras2vox, coordinates)
+
+    # Round indices to nearest integer
+    # The file standard assumes that the voxel coordinates refer to the center of each voxel.
+    # https://brainder.org/2012/09/23/the-nifti-file-format/
+    ijk_rounded = np.rint(ijk_vectorized).astype("int")
```

Comment:
Do you know if these values are integers (i.e `1.0`, `2.0` etc) or if the the operations performed actually makes the values floats? I am just thinking about the scenario when you have two neighbouring voxels with very different labels. Or is this perhaps not a relevant concern in this case?

Reply:
This relates to voxel indices, not the values:)